applications/IgaApplication/custom_elements/shell_6p_element.h

```diff
@@ -0,0 +1,667 @@
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Calculates LHS and RHS dependent on flags
+    void CalculateAll(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo,
+        const bool CalculateStiffnessMatrixFlag,
+        const bool CalculateResidualVectorFlag
+    );
+
+    /// Initialize Operations
+    void InitializeMaterial();
+
+    void CalculateKinematics(
+        const IndexType IntegrationPointIndex,
+        KinematicVariables& rKinematicVariables) const;
+
+    // Computes transformation
+    void CalculateTransformation(
+        const KinematicVariables& rKinematicVariables,
+        Matrix& rT) const;
+    
+    // Computes transformation for the stress tensor 
+    void CalculateTransformationFromCovariantToCartesian(
+        const KinematicVariables& rKinematicVariables,
+        Matrix& rTCovToCar) const;
+
+    void CalculateB(
+        const IndexType IntegrationPointIndex,
+        Matrix& rB,
+        double zeta,
+        Matrix& DN_De_Jn,
+        Matrix& J_inv,
+        Matrix& dn,
+        const KinematicVariables& rActualKinematic) const;
+
+    void CalculateBGeometric(
+        const IndexType IntegrationPointIndex,
+        Matrix& rB,
+        double zeta,
+        Matrix& DN_De_Jn,
+        Matrix& J_inv,
+        Matrix& dn,
+        const KinematicVariables& rActualKinematic) const;
+
+    void CalculateJn(
+        const IndexType IntegrationPointIndex,
+        KinematicVariables& rKinematicVariables,
+        double zeta,
+        Matrix& DN_De_Jn,
+        Matrix& J_inv,
+        Matrix& dn,
+        double& area) const;
+
+    void CalculateBDrill(
+        const IndexType IntegrationPointIndex,
+        Matrix& rBd,
+        Matrix& DN_De_Jn,
+        const KinematicVariables& rActualKinematic) const;
+
+    void CalculateStressMatrix(
+        array_1d<double, 6> stress_vector,
+        Matrix& stress_matrix
+    ) const;
+
+    void CalculateSecondVariationStrainCurvature(
+        const IndexType IntegrationPointIndex,
+        SecondVariations& rSecondVariationsStrain,
+        SecondVariations& rSecondVariationsCurvature,
+        const KinematicVariables& rActualKinematic) const;
+
+    /**
+    * This functions updates the constitutive variables
+    * @param rActualMetric: The actual metric
+    * @param rThisConstitutiveVariables: The constitutive variables to be calculated
+    * @param rValues: The CL parameters
+    * @param ThisStressMeasure: The stress measure considered
+    */
+    void CalculateConstitutiveVariables(
+        const IndexType IntegrationPointIndex,
+        KinematicVariables& rActualMetric,
+        ConstitutiveVariables& rThisConstitutiveVariables,
+        ConstitutiveLaw::Parameters& rValues,
+        const ConstitutiveLaw::StressMeasure ThisStressMeasure
+    ) const;
+
+    inline void CalculateAndAddK(
```

Comment:
[`inline`](https://en.cppreference.com/w/cpp/language/inline.html) does not do what you think it does. It's completely unnecessary here.

```suggestion
    void CalculateAndAddK(
```

Also these functions should all be moved to source.